[qtAliceVision] FloatImageViewer: Set the loading status to "error" when an image fails to be loaded

cbentejac opened the pull request:
This PR does the following:
1. Add an enum `error` member to the `ResponseData` structure that is meant to be set whenever an error occurs while attempting to load an image. "Missing file" errors are distinguished from generic errors (and in particular, from "Can't open file" errors).
2. Add an enum that reflects the global status of the FloatImageViewer. It describes accurately whether the viewer has encountered an error (with several types of errors, which can be extended as needed), if an image is being loaded, or if nothing is happening (and no error has been detected). The global status uses the `loading` status to determine its current state. As a consequence, the `loading` status is not exposed anymore to the QML side.
The global status can currently be set with the following values:
    - `NONE`: nothing is happening and the last operations have been successful;
    - `LOADING`: an image is being loaded (the `loading` status is set to true);
    - `OUTDATED_LOADING`: an image was already loading during the previous reload() operation (the `loading` status is still set to true but something wrong might be happening somewhere);
    - `MISSING_FILE`: the file to load is missing;
    - `ERROR`: an unspecified error occurred.

This relates to the following Meshroom pull request: https://github.com/alicevision/Meshroom/pull/2250.

cbentejac committed:
[qtAliceVision] SingleImageLoader: Catch errors when loading image

If an exception is caught when attempting to load an image, instead
of just logging the error message and keeping on going, a flag is set
in the `ResponseData` object.

The flag's value depends on the type of exception/error that is caught.
For now, there are only two types of errors: either the file cannot be found
(`MISSING_FILE`) or cannot be opened (`ERROR`).

fabiencastan reviewed (changes requested):
Use a general STATUS instead of LoadingStatus.

cbentejac committed:
[qtAliceVision] FloatImageViewer: Add a global status

Instead of only having a loading status that oscillates between the "loading"
and "not loading" state, add a global status for the viewer that can describe
more states.

The global status is updated depending on the loading status and on the error
flag of the `ResponseData` object.

The loading status itself is kept as it allows to distinguish several
loading use cases but is not exposed to the QML anymore, as the global status
covers all the relevant information.

fabiencastan commented on src/qtAliceVision/SingleImageLoader.cpp:
```diff
@@ -90,6 +90,16 @@ void SingleImageLoadingIORunnable::run()
     catch (const std::runtime_error& e)
     {
+        // std::runtime_error at this point is a "can't find/open image" error
+        if (access(_reqData.path.c_str(), F_OK) == -1)  // "can't find image" case
```
std::filesystem::exists

fabiencastan replied:
in other places, we currently use boost::filesystem::exists as it was before c++17.

cbentejac replied:
Done.
